ptpython/python_input.py

```diff
@@ -153,40 +154,42 @@ def __init__(self,
         self._extra_buffer_processors = _extra_buffer_processors or []
 
         # Settings.
-        self.show_signature = True
-        self.show_docstring = False
-        self.show_meta_enter_message = True
-        self.completion_visualisation = CompletionVisualisation.MULTI_COLUMN
-        self.completion_menu_scroll_offset = 1
-
-        self.show_line_numbers = False
-        self.show_status_bar = True
-        self.wrap_lines = True
-        self.complete_while_typing = True
-        self.vi_mode = vi_mode
-        self.paste_mode = False  # When True, don't insert whitespace after newline.
-        self.confirm_exit = True  # Ask for confirmation when Control-D is pressed.
-        self.accept_input_on_enter = 2  # Accept when pressing Enter 'n' times.
-                                        # 'None' means that meta-enter is always required.
-        self.enable_open_in_editor = True
-        self.enable_system_bindings = True
-        self.enable_input_validation = True
-        self.enable_auto_suggest = False
-        self.enable_mouse_support = False
-        self.enable_history_search = False  # When True, like readline, going
-                                            # back in history will filter the
-                                            # history on the records starting
-                                            # with the current input.
-
-        self.highlight_matching_parenthesis = True
-        self.show_sidebar = False  # Currently show the sidebar.
-        self.show_sidebar_help = True # When the sidebar is visible, also show the help text.
-        self.show_exit_confirmation = False  # Currently show 'Do you really want to exit?'
-        self.terminal_title = None  # The title to be displayed in the terminal. (None or string.)
-        self.exit_message = 'Do you really want to exit?'
-
-        # Tokens to be shown at the prompt.
-        self.prompt_style = 'classic'  # The currently active style.
+        self.settings = Settings({
+            'show_signature': True,
```

Comment:
A better approach could allow settings to be made in the repl object (repl.show_sidebar, etc, as in old config.py modules) to do not break backwards compatibility, provide a __setattr__ in  PythonInput to move those to the settings attribute, and display a warning FutureWarning to notify the change in the next release.